source/common/network/utility.cc

```diff
@@ -356,6 +356,17 @@ Address::InstanceConstSharedPtr Utility::getOriginalDst(Socket& sock) {
   }
 
   if (status != 0) {
+#ifdef IP_TRANSPARENT
```

Comment:
I thought we needn't this since we already use `Api::OsSysCallsSingleton::get().supportsIpTransparent()`

Reply:
I've got compiling error of IP_TRANSPARENT not define in msvc ci building... Should I use another macro to set these lines apart from msvc platform ?

Reply:
Maybe worth a try on this
https://github.com/envoyproxy/envoy/blob/15d8b93608bc5e28569f8b042ae666a5b09b87e9/source/common/network/socket_option_impl.h#L14-L24

Also looks like you need to take care of different IP version address.

Reply:
thanks for your help, I will improve this in next commit